### DISCUSSION TITLE
Support only ESM or CJS

### DISCUSSION THREAD

**Discussion by coreybutler** [coreybutler/node-windows#309] [Ideas]
The library is showing its age and could really use a significant refactor. There are a few options going forward, and I'd like to get user feedback before taking action.

This module is one of the only ones I maintain that still uses CommonJS (i.e. `require`). I would prefer to move entirely to ES modules, but I realize this may affect existing applications. I'd like to know how many people still need to support CommonJS.

I'm considering separating this project into two variants: ESM (node-windows-esm) and CJS (node-windows). The CJS version would essentially be treated as a legacy edition, meaning I'd likely be seeking a maintainer once the initial work is done.

I'm seeking community feedback.

Thanks in advance,

Corey

**Comment by DevRCRun** [comment]
To get this straight in my head, I think we're talking about our customised install / uninstall scripts needing to move across from CJS to ESM here, and not the applications themselves monitored by node-windows?

We haven't had a chance to look at moving away from CJS ourselves across the board, but our node-windows scripts may be a relatively self contained place to start...

**Reply by coreybutler** [reply]
Correct, this is just for install/uninstall scripts that use node-windows. This change would not affect the applications installed with node-windows.

**Comment by nir-moshe** [comment]
keep CJS support please

**Comment by wuyiwei** [comment]
keep CJS support please

**Comment by scdba** [comment]
My 2p
Splitting into ESM and CJS and CJS considered legacy makes sense.
Especially if ESM eases the work on resolving https://github.com/coreybutler/node-windows/issues/308.
Those of us on CJS will just have to live with beta 6 or do the work to move to ESM (that we are putting off ;))

**Comment by coreybutler** [comment]
I've decided to leave the library "as is" for now while I [contemplate building an entirely new service management library](https://github.com/coreybutler/node-windows/discussions/317).
